src/main/java/frc/trigon/robot/subsystems/elevator/Elevator.java

```diff
@@ -92,21 +101,25 @@ void setTargetState(ElevatorConstants.ElevatorState targetState) {
     }
 
     void setTargetPosition(double targetPositionMeters, double speedPercentage) {
-        elevatorIO.setTargetPosition(toRevolutions(targetPositionMeters), speedPercentage);
+        motor.setControl(scaleProfile(positionRequest.withPosition(targetPositionMeters), speedPercentage));
     }
 
     private void updateNetworkTables() {
         updateMechanism();
         Logger.recordOutput("Elevator/ElevatorPositionMeters", getPositionMeters());
-        Logger.recordOutput("Elevator/ElevatorVelocityMetersPerSecond", toMeters(elevatorInputs.velocityRevolutionsPerSecond));
+        Logger.recordOutput("Elevator/ElevatorVelocityMetersPerSecond", toMeters(motor.getSignal(TalonFXSignal.VELOCITY)));
+    }
+
+    private DynamicMotionMagicVoltage scaleProfile(DynamicMotionMagicVoltage profile, double speedPercentage) {
+        return profile.withVelocity(ElevatorConstants.MOTION_MAGIC_CRUISE_VELOCITY * (speedPercentage / 100)).withAcceleration(ElevatorConstants.MOTION_MAGIC_ACCELERATION * (speedPercentage / 100));
     }
 
     private void updateMechanism() {
-        ElevatorConstants.ELEVATOR_LIGAMENT.setLength(toMeters(elevatorInputs.positionRevolutions) + ElevatorConstants.RETRACTED_ELEVATOR_LENGTH_METERS);
-        ElevatorConstants.TARGET_ELEVATOR_POSITION_LIGAMENT.setLength(toMeters(elevatorInputs.profiledSetpointRevolutions) + ElevatorConstants.RETRACTED_ELEVATOR_LENGTH_METERS);
+        ElevatorConstants.MECHANISM.setCurrentPosition(getPositionMeters() + ElevatorConstants.RETRACTED_ELEVATOR_LENGTH_METERS);
+        ElevatorConstants.MECHANISM.setTargetPosition(toMeters(motor.getSignal(TalonFXSignal.CLOSED_LOOP_REFERENCE)) + ElevatorConstants.RETRACTED_ELEVATOR_LENGTH_METERS);
```

Comment:
this can also be accepted in the `update` function.
no need for `+ retracted`, the elevator mech does that by default now